### PR TITLE
DEPR: Enforce deprecation of groupby(..., axis=1)

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Enforcing #51203

### CODE REVIEW

File: doc/source/whatsnew/v3.0.0.rst [hunk 1]
```diff
@@ -102,8 +102,10 @@ Deprecations
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 - Removed :meth:`DataFrameGroupby.fillna` and :meth:`SeriesGroupBy.fillna` (:issue:`55719`)
+- Removed ``axis`` argument from :meth:`DataFrame.groupby` and :meth:`Series.groupby` (:issue:`51203`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:105]: Could you mention rolling and resample too?

Comment (rhshadrach) [reply doc/source/whatsnew/v3.0.0.rst]: Thanks, done.

File: pandas/core/groupby/generic.py [hunk 2]
```diff
@@ -1543,16 +1519,11 @@ def _cython_transform(
...
         numeric_only: bool = False,
-        axis: AxisInt = 0,
         **kwargs,
     ) -> DataFrame:
-        assert axis == 0  # handled by caller
-
-        # With self.axis == 0, we have multi-block tests
+        # We have multi-block tests
         #  e.g. test_rank_min_int, test_cython_transform_frame
         #  test_transform_numeric_ret
```
[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/core/groupby/generic.py:1526]: I think so too. Can be done in a follow up

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach

**Pull request merged**